collect/collect_test.go

```diff
@@ -480,6 +480,10 @@ func TestAddSpan(t *testing.T) {
 
 	coll.AddSpanFromPeer(spanFromPeer)
 
+	// We need to wait for the span to be processed manually with a Sleep
+	// because trace object is not concurrent safe
+	time.Sleep(conf.GetTracesConfig().GetSendTickerValue() * 2)
```

Comment:
Sleeps make me sad. The `EventuallyWithT()` didn't work for a wait?

Reply:
Unfortunately, `EventuallyWithT` launches an goroutine to do its work. That can cause race condition with the collect loop when adding a span onto the trace object